symphony-bdk-core/src/main/java/com/symphony/bdk/core/config/BdkConfigParser.java

```diff
@@ -27,10 +28,12 @@
 
     private static final ObjectMapper JSON_MAPPER = new JsonMapper();
     private static final ObjectMapper YAML_MAPPER = new YAMLMapper();
+    private static final StringSubstitutor envVarStringSubstitutor;
 
     static {
         JSON_MAPPER.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
         YAML_MAPPER.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        envVarStringSubstitutor = new StringSubstitutor(StringLookupFactory.INSTANCE.environmentVariableStringLookup());
```

Comment:
In the doc they mention that StringSubstitutor is not thread safe, could it be a problem?

maybe to be safe I would change BdkConfigParser to create an instance when parsing a config instead of having it as a static utils

Reply:
I found that: https://stackoverflow.com/a/2412144

Reply:
the variable will be thread safe in the sense it will be created only once but the usage of StringSubstitutor by multiple threads won't

Reply:
ok I moved it to instance-based system